src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1157,7 +1157,8 @@ static bool processContextAttributeVector (ContextElement*
                                            double&                                    coordLat,
                                            double&                                    coordLong,
                                            std::string                                tenant,
-                                           const std::vector<std::string>&            servicePathV)
+                                           const std::vector<std::string>&            servicePathV,
+                                           std::string*                               why)
```

Comment:
Probably is more efficient to have an integer for that, considering that the number of "why" contitions is finite and small. Moreover, I'd sugget to pass this as a reference (as it is our style in other places, e.g. the `caller` argument added to mongoUpdateContext() function.

```
#define #ATTRIBUTE_NOT_FOUND (in the right .h file)
...
why = ATTRIBUTE_NOT_FOUND
```